modules/lang-painless/src/main/java/org/elasticsearch/painless/Def.java

```diff
@@ -460,46 +395,28 @@ static MethodHandle lookupGetter(PainlessLookup painlessLookup, Class<?> receive
      */
     static MethodHandle lookupSetter(PainlessLookup painlessLookup, Class<?> receiverClass, String name) {
         // first try whitelist
-        for (Class<?> clazz = receiverClass; clazz != null; clazz = clazz.getSuperclass()) {
-            PainlessClass struct = painlessLookup.lookupPainlessClass(clazz);
-
-            if (struct != null) {
-                MethodHandle handle = struct.setterMethodHandles.get(name);
-                if (handle != null) {
-                    return handle;
+        try {
+            return painlessLookup.lookupRuntimeSetterMethodHandle(receiverClass, name);
+        } catch (IllegalArgumentException iae) {
```

Comment:
I don't think we should use IAE as control flow. Can we return null if the the class is not found and base these special cases on that?

Reply:
Fixed.